torch/distributed/device_mesh.py

```diff
@@ -1001,11 +1001,33 @@ def ndim(self) -> int:
         def shape(self) -> tuple[int, ...]:
             return tuple(self.mesh.shape)
 
-        def get_rank(self) -> int:
+        def get_rank(self, coordinate: Optional[list[int]] = None) -> int:
```

Comment:
hmmm can we not mix it with this API? I think they are doing two different things, why are we doing this piggyback? Also, I am wondering if this function can be inside DTensor.

One more thing, since now device mesh is using layout, any chance you can directly use `crd2idx` inside torch/distributed/_pycute/int_tuple.py? That way you don't really need this code change.